packages/legend-query/src/stores/QueryBuilderExplorerState.ts

```diff
@@ -305,30 +335,35 @@ export const getPropertyNodeMappingData = (
         graphManagerState,
         parentTargetSetImpl,
       );
+      const propertyMappingsFromSuperSetImpls = superSetImplementations
+        .map((s) => resolvePropertyMappingsForSetImpl(graphManagerState, s))
+        .flat();
       const mappedProperties = propertyMappings
         .filter((p) => !p.isStub)
         .map((p) => p.property.value.name);
+      const mappedPropertiesFromSuperSetImpls =
+        propertyMappingsFromSuperSetImpls
+          .filter((p) => !p.isStub)
+          .map((p) => p.property.value.name);
+
       // check if property is mapped through defined property mappings
-      if (mappedProperties.includes(property.name)) {
+      if (
+        mappedProperties.includes(property.name) ||
+        mappedPropertiesFromSuperSetImpls.includes(property.name)
+      ) {
         // if class we need to resolve the Set Implementation
         if (property.genericType.value.rawType instanceof Class) {
-          const propertyMapping = propertyMappings.find(
-            (p) => p.property.value === property,
-          );
-          if (propertyMapping) {
-            const targetSetImpl =
-              resolveTargetSetImplementationForPropertyMapping(propertyMapping);
-            return {
-              mapped: true,
-              // NOTE: we could potentially resolve all the leaves and then overlap them somehow to
-              // help identifying the mapped properties. However, we would not do that here
-              // as opertion mapping can support more complicated branching logic (right now we just assume
-              // it's always simple union), that Studio should not try to analyze.
-              skipMappingCheck:
-                targetSetImpl instanceof OperationSetImplementation,
-              targetSetImpl,
-            };
+          let propertyMappingArray;
+          // propertyMappings from the current level is the highest priority
+          if (mappedProperties.includes(property.name)) {
+            propertyMappingArray = propertyMappings;
+          } else {
+            propertyMappingArray = propertyMappingsFromSuperSetImpls;
           }
+          return generatePropertyNodeMappingDataForNonPrimitveProperty(
+            property,
```

Comment:
Lets keep the logic as is and only `resolveTargetSetImplementationForPropertyMapping` if a propertyMapping exists (remove the guaranteNonNullable above please :))

Reply:
there is if condition above which checks if the mappedProperties or mappedPropertiesFromSuperSetImpls contains property name, ideally if this returns true, a propertyMapping exits correspondingly.